Fix a CMS test that fails in nightly build

daxian-dbw opened the pull request.

mirichmo commented on test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1:
```diff
@@ -271,16 +271,28 @@ Describe "CmsMessage cmdlets thorough tests" -Tags "Feature" {
+            {
+                $goodCerts = @(Get-ChildItem Cert:\currentuser\My -DocumentEncryptionCert | ForEach-Object Thumbprint)
+                $badCert = Get-ChildItem Cert:\currentuser\My | ? { $_.Thumbprint -notin $goodCerts } | Select -First 1
+                $skipTest = $badCert -eq $null ## No proper Cert found.
```
Does this indicate a failure during setup? I'm a little worried that this line will silently hide a setup failure that won't be noticed because the test will fall into the "skipped" bucket.

iSazonov replied:
Maybe better not skip but "Should" in the test below?

daxian-dbw replied:
No, it's not a failure during setup. CI builds are unlike our dev machines, there is just no certs under `Cert:\currentuser\My`

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1:
```diff
@@ -271,16 +271,28 @@ Describe "CmsMessage cmdlets thorough tests" -Tags "Feature" {
+            if ($IsWindows)
+            {
+                $goodCerts = @(Get-ChildItem Cert:\currentuser\My -DocumentEncryptionCert | ForEach-Object Thumbprint)
+                $badCert = Get-ChildItem Cert:\currentuser\My | ? { $_.Thumbprint -notin $goodCerts } | Select -First 1
```
Where these certificates come from?

daxian-dbw replied:
Unlike my dev machines, it seems AppVeyor builders don't have any certs in `cert:\currentuser\my`. I probably should generate a bad cert for this test.

daxian-dbw replied:
I added a certificate that cannot be used for data encryption, so as to verify the error condition.

iSazonov replied:
#Closed.

daxian-dbw committed:
Fix a CMS test that fails in nightly build.
Add a certificate that cannot be used to encrypt data to verify an error condition.

iSazonov commented:
LGTM.

mirichmo commented on test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1:
```diff
@@ -124,9 +151,12 @@ Describe "CmsMessage cmdlets thorough tests" -Tags "Feature" {
+
             if ($IsCoreCLR)
             {
                 # PKI module is not available for PowerShell Core, so we need to use Windows PowerShell to import the cert
```
This test cannot be run on environments other than full Windows PowerShell SKUs. What about skipping the test if the test detects that it is running on Nano Server or IoT? Is that beyond the scope of our testing?

daxian-dbw replied:
I have verified that it works fine on NanoServer, so thus I assume the same on IoT. Here is a screenshot:

![image](https://cloud.githubusercontent.com/assets/127450/23675916/c03f3686-032f-11e7-8344-dcfbefbf0174.png)
